var/ramble/repos/builtin/applications/osu-micro-benchmarks/application.py

```diff
@@ -25,43 +30,347 @@ class OsuMicroBenchmarks(SpackApplication):
+                                 units='us',
+                                 contexts=['msg_size'])
+            self.figure_of_merit('P99 Tail Latency',
+                                 fom_regex=fom_regex,
+                                 group_name='p99',
+                                 units='us',
+                                 contexts=['msg_size'])
+        elif fom_type == self.fom_types.single_bw:
+            fom_regex = r'\s*(?P<msg_size>[0-9]+)\s+(?P<bw>[0-9\.]+)\s*$'
+            self.figure_of_merit('Bandwidth',
+                                 fom_regex=fom_regex,
+                                 group_name='bw',
+                                 units='MB/s',
+                                 contexts=['msg_size'])
+        elif fom_type == self.fom_types.single_tail_bw:
+            fom_regex = r'\s*(?P<msg_size>[0-9]+)\s+(?P<avg>[0-9\.]+)\s+(?P<p50>[0-9\.]+)\s+(?P<p90>[0-9\.]+)\s+(?P<p99>[0-9\.]+)\s*$'
+            self.figure_of_merit('Average Bandwidth',
+                                 fom_regex=fom_regex,
+                                 group_name='avg',
+                                 units='MB/s',
+                                 contexts=['msg_size'])
+            self.figure_of_merit('P50 Tail Bandwidth',
+                                 fom_regex=fom_regex,
+                                 group_name='p50',
+                                 units='MB/s',
+                                 contexts=['msg_size'])
+            self.figure_of_merit('P90 Tail Bandwidth',
+                                 fom_regex=fom_regex,
+                                 group_name='p90',
+                                 units='MB/s',
+                                 contexts=['msg_size'])
+            self.figure_of_merit('P99 Tail Bandwidth',
+                                 fom_regex=fom_regex,
+                                 group_name='p99',
+                                 units='MB/s',
+                                 contexts=['msg_size'])
+        elif fom_type == self.fom_types.multi_bw:
+            fom_regex = r'\s*(?P<msg_size>[0-9]+)\s+(?P<bw>[0-9\.]+)\s+(?P<msg_rate>[0-9\.]+)\s*$'
+            self.figure_of_merit('Bandwidth',
+                                 fom_regex=fom_regex,
+                                 group_name='bw',
+                                 units='MB/s',
+                                 contexts=['msg_size'])
+            self.figure_of_merit('Message Rate',
+                                 fom_regex=fom_regex,
+                                 group_name='msg_rate',
+                                 units='Messages/s',
+                                 contexts=['msg_size'])
+        elif fom_type == self.fom_types.multi_tail_bw:
+            fom_regex = r'\s*(?P<msg_size>[0-9]+)\s+(?P<bw>[0-9\.]+)\s+(?P<msg_rate>[0-9\.]+)\s+(?P<p50>[0-9\.]+)\s+(?P<p90>[0-9\.]+)\s+(?P<p99>[0-9\.]+)\s*$'
+            self.figure_of_merit('Bandwidth',
```

Comment:
We repeat a lot of these FOM blocks, can we DRY them out?

Reply:
How about the newer commit?